test/lib/pipeline.test.js

```diff
@@ -299,6 +305,32 @@ describe('Pipeline Model', () => {
         });
     });
 
+    describe('get secrets', () => {
+        it('has a secrets getter', () => {
+            const listConfig = {
+                params: {
+                    pipelineId: pipeline.id
+                },
+                paginate: {
+                    count: 25, // This limit is set by the matrix restriction
+                    page: 1
+                }
+            };
+
+            secretFactoryMock.list.resolves(null);
+            // when we fetch secrets it resolves to a promise
+            assert.isFunction(pipeline.secrets.then);
+            // and a factory is called to create that promise
+            assert.calledWith(secretFactoryMock.list, listConfig);
+
+            // When we call pipeline.secrets again it is still a promise
+            assert.isFunction(pipeline.secrets.then);
+            // ...but the factory was not recreated, since the promise is stored
+            // as the model's pipeline property, now
```

Comment:
I looked back and saw that `pipeline.secrets` is an attribute an not a function:

```
const whatever = pipeline.secrets;

return whatever.then(() => {
    assert.calledWith(secretFactoryMock.list, listConfig);

    return pipeline.secrets;
})
.then(() => {
    assert.calledOnce(secretFactoryMock.list);
});
```